internal/lang/funcs/encoding.go

```diff
@@ -178,6 +179,37 @@ var Base64GzipFunc = function.New(&function.Spec{
 	},
 })
 
+// GunzipBase64Func constructs a function that Bae64 decodes a string and decompresses the result with gunzip.
+var GunzipBase64Func = function.New(&function.Spec{
+	Params: []function.Parameter{
+		{
+			Name: "str",
+			Type: cty.String,
+		},
+	},
+	Type:         function.StaticReturnType(cty.String),
+	RefineResult: refineNotNull,
+	Impl: func(args []cty.Value, retType cty.Type) (cty.Value, error) {
+		str, strMarks := args[0].Unmark()
+		s := str.AsString()
+		sDec, err := base64.StdEncoding.DecodeString(s)
+		if err != nil {
+			return cty.UnknownVal(cty.String), fmt.Errorf("failed to decode base64 data %s", redactIfSensitive(s, strMarks))
+		}
+		sDecBuffer := bytes.NewReader(sDec)
+		gzipReader, err := gzip.NewReader(sDecBuffer)
+		if err != nil {
+			return cty.UnknownVal(cty.String), fmt.Errorf("failed to gunzip base64 decoded data: %w", err)
```

Comment:
**(nitpick)** The data are no longer base64-encoded, hence no need to mention that in the error message IMO.

```suggestion
                        return cty.UnknownVal(cty.String), fmt.Errorf("failed to gunzip bytestream: %w", err)
```

Reply:
done